prov/hook/hook_hmem/src/hook_hmem.c

```diff
@@ -81,7 +81,7 @@ static int hook_hmem_add_region(struct hook_hmem_domain *domain,
 	attr.access = FI_SEND | FI_RECV | FI_READ | FI_WRITE |
 		      FI_REMOTE_READ | FI_REMOTE_WRITE;
 	attr.offset = 0;
-	attr.requested_key = (uint64_t) *hmem_desc;
+	attr.requested_key = (uintptr_t) *hmem_desc;
```

Comment:
Here we are indeed expecting `uint64_t`. The conversion works for both 32-bit and 64-bit machines is `(uint64_t)(uintptr_t) *hmem_desc`